.github/workflows/unit_tests.yaml

```diff
@@ -0,0 +1,178 @@
+name: "Unit tests"
```

Comment:
is there any way that we can have like a parameterized workflow? would be nice not to repeat the stuff from the integration tests here

Reply:
So I spent some time looking into this (and I did turn the cache logic into a reusable action) and tried to do the same with the tests, but that won't work for annoying reasons.

1. For the windows tests we use powershell and for mac/linux we use bash, which was nbd bc I could have a run step for the windows tests and one for mac/linux and then "if" them based on the platform
2. "if" isn't supported on composite actions so then I found a hack where I could do an "if" in the command logic and if we were doing the windows tests on a non-windows machine, then just exit 0.
3. But the retry action we are using doesn't let you use the powershell shell on non-windows executors, so that step would always fail.

So basically it was hacks on hacks so I decided to stop investigating. I think perhaps we could make the windows tests run on bash instead of using powershell but I've sorta already spent 1.5 days on this and since we'll be revisiting the actions to add the release steps, I can do more investigation then.

Reply:
no worries, let's leave this as it is then and we'll figure something out :) this is a great start